irksome/dirk_stepper.py

```diff
@@ -101,31 +100,30 @@ def getFormDIRK(F, butch, t, dt, u0, bcs=None):
     stage_F = replace(F, repl)
 
     bcnew = []
-    gblah = []
 
     # For the DIRK case, we need one new BC for each old one (rather
     # than one per stage), but we need a `Function` inside of each BC
     # and a rule for computing that function at each time for each
     # stage.
+    a_vals = numpy.array([MC.Constant(0) for i in range(num_stages)],
+                         dtype=object)
+    d_val = MC.Constant(1.0)
     for bc in bcs:
         Vbc = bc.function_space()
+        dat4bc = getThingy(V, bc)
         bcarg = as_ufl(bc._original_arg)
         bcarg_stage = replace(bcarg, {t: t+c*dt})
-        try:
-            gdat = assemble(interpolate(bcarg, Vbc))
-            gmethod = lambda gd, gc: gd.interpolate(gc)
-        except:  # noqa: E722
-            gdat = assemble(project(bcarg, Vbc))
-            gmethod = lambda gd, gc: gd.project(gc)
+
+        gdat = bcarg_stage - dat4bc(u0)
+        for i in range(num_stages):
+            gdat -= dt*a_vals[i]*dat4bc(ks[i])
+
+        gdat /= dt*d_val
 
         new_bc = DirichletBC(Vbc, gdat, bc.sub_domain)
         bcnew.append(new_bc)
 
-        dat4bc = getThingy(V, bc)
-        gdat2 = Function(gdat.function_space())
-        gblah.append((gdat, gdat2, bcarg_stage, gmethod, dat4bc))
-
-    return stage_F, (k, g, a, c), bcnew, gblah
+    return stage_F, (k, g, a, c, a_vals, d_val), bcnew
```

Comment:
Are we returning `(a_vals, d_val)` instead of `gblah`? Is there a good reason to merge these into the original `kgac` tuple?

Reply:
My logic was that these are generalizations of the `a` and `c` Constants that we're already returning here, but I agree that the naming is ugly

Reply:
Fixed now